internal/api/api.go

```diff
@@ -2,42 +2,99 @@ package api
 
 import (
 	"context"
+	"github.com/ozoncp/ocp-team-api/internal/models"
+	"github.com/ozoncp/ocp-team-api/internal/repo"
 	desc "github.com/ozoncp/ocp-team-api/pkg/ocp-team-api"
-	log "github.com/rs/zerolog/log"
+	"github.com/rs/zerolog/log"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 type api struct {
 	desc.UnimplementedOcpTeamApiServer
+	repo repo.Repo
 }
 
-func NewOcpTeamApi() desc.OcpTeamApiServer {
-	return &api{}
+func NewOcpTeamApi(repo repo.Repo) desc.OcpTeamApiServer {
+	return &api{
+		repo: repo,
+	}
 }
 
 func (a *api) CreateTeamV1(
 	ctx context.Context,
 	req *desc.CreateTeamV1Request) (*desc.CreateTeamV1Response, error) {
 	log.Printf("Create team (name=%s, description=%s)", req.Name, req.Description)
-	return &desc.CreateTeamV1Response{}, nil
+
+	id, err := a.repo.AddTeam(ctx, models.Team{Name: req.Name, Description: req.Description})
+
+	if err != nil {
+		log.Info().Err(err)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	log.Info().Msgf("new team was created successfully with id=%d", id)
+
+	return &desc.CreateTeamV1Response{Id: id}, nil
 }
 
 func (a *api) GetTeamV1(
 	ctx context.Context,
 	req *desc.GetTeamV1Request) (*desc.GetTeamV1Response, error) {
 	log.Printf("Get team (id=%d)", req.Id)
-	return &desc.GetTeamV1Response{}, nil
+
+	team, err := a.repo.GetTeam(ctx, req.Id)
+
+	if err != nil {
+		log.Info().Err(err)
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	response := &desc.GetTeamV1Response{
+		Team: &desc.Team{
+			Id:          team.Id,
+			Name:        team.Name,
+			Description: team.Description,
+		},
+	}
+
+	return response, nil
 }
 
 func (a *api) ListTeamsV1(
 	ctx context.Context,
 	req *desc.ListTeamsV1Request) (*desc.ListTeamsV1Response, error) {
 	log.Printf("List teams (limit=%d, offset=%d)", req.Limit, req.Offset)
-	return &desc.ListTeamsV1Response{}, nil
+
+	teams, err := a.repo.ListTeams(ctx, req.Limit, req.Offset)
+
+	if err != nil {
+		log.Info().Err(err)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	responseTeams := make([]*desc.Team, len(teams), cap(teams))
+	for i, team := range teams {
+		responseTeams[i] = &desc.Team{
+			Id:          team.Id,
+			Name:        team.Name,
+			Description: team.Description,
+		}
+	}
```

Comment:
Ты написал правильно, но в таком режиме очень просто выстрелить себе в член:) Как минимум потому, что ты создаёшь слайс заданного размера с дефолтными значениями – далеко не всегда на выходе нужен такой. Можно написать, не заметить – и ловить ошибки потом (ожидаешь слайс с 10 объектами, получаешь – с 15, из них 5 последних – дефолтные). Плюс ты капэсити инициализируешь по капэсити `teams` – но у `teams` оно может быть больше, чем тебе нужно! У тебя здесь не будет никогда элементов больше, чем в `teams`, поэтому надо задавать капэсити по длине `teams`

Есть вариант, который под капотом такой же, но менее опасный, рекомендую в основном его придерживаться (кроме тех случаев, где ты реально понимаешь, почему лучше сделать иначе):
```suggestion
	responseTeams := make([]*desc.Team, 0, len(teams))
	for _, team := range teams {
		responseTeams = append(responseTeams, &desc.Team{
			Id:          team.Id,
			Name:        team.Name,
			Description: team.Description,
		})
	}
```

Reply:
cap(teams) - я не хотел этого :) протупил
поправил